src/jit/flowgraph.cpp

```diff
@@ -20588,6 +20588,270 @@ void Compiler::fgStress64RsltMul()
     fgWalkAllTreesPre(fgStress64RsltMulCB, (void*)this);
 }
 
+// BBPredsChecker checks jumps from the block's predecessors to the block.
+class BBPredsChecker
+{
+public:
+    BBPredsChecker(Compiler* compiler);
+
+    unsigned CheckBBPreds(BasicBlock* block, unsigned curTraversalStamp);
+
+private:
+    bool CheckEhTryDsc(BasicBlock* block, BasicBlock* blockPred, EHblkDsc* ehTryDsc);
+    bool CheckEhHndDsc(BasicBlock* block, BasicBlock* blockPred, EHblkDsc* ehHndlDsc);
+    bool CheckJump(BasicBlock* blockPred, BasicBlock* block);
+    bool CheckEHFinalyRet(BasicBlock* blockPred, BasicBlock* block);
+
+private:
+    Compiler* comp;
+};
+
+BBPredsChecker::BBPredsChecker(Compiler* compiler) : comp(compiler)
```

Comment:
I'd make this inline. In general, placing member definitions outside of the class is of little use if the class is not in a header file.

Reply:
I believe it makes it harder to scan the API of the class, doesn't it?
I think it is a matter of taste, but if more people agree that it is better to inline these functions I would not mind.

Reply:
Makes sense, fixed for the constructor. Thanks.